In [1]:
import kagglehub
path = kagglehub.dataset_download("carrie1/ecommerce-data")

Using Colab cache for faster access to the 'ecommerce-data' dataset.


In [5]:
!pip install kagglehub


In [6]:
import kagglehub

path = kagglehub.dataset_download("carrie1/ecommerce-data")

print("Dataset downloaded to:", path)


Using Colab cache for faster access to the 'ecommerce-data' dataset.
Dataset downloaded to: /kaggle/input/ecommerce-data


In [7]:
import os

print(os.listdir(path))


['data.csv']


In [3]:
import pandas as pd
import numpy as np

In [9]:
import pandas as pd

file_path = os.path.join(path, "data.csv")
df = pd.read_csv(file_path, encoding='latin1')

df.head()


InvoiceNo StockCode                          Description  Quantity  \
0    536365    85123A   WHITE HANGING HEART T-LIGHT HOLDER         6   
1    536365     71053                  WHITE METAL LANTERN         6   
2    536365    84406B       CREAM CUPID HEARTS COAT HANGER         8   
3    536365    84029G  KNITTED UNION FLAG HOT WATER BOTTLE         6   
4    536365    84029E       RED WOOLLY HOTTIE WHITE HEART.         6   

      InvoiceDate  UnitPrice  CustomerID         Country  
0  12/1/2010 8:26       2.55     17850.0  United Kingdom  
1  12/1/2010 8:26       3.39     17850.0  United Kingdom  
2  12/1/2010 8:26       2.75     17850.0  United Kingdom  
3  12/1/2010 8:26       3.39     17850.0  United Kingdom  
4  12/1/2010 8:26       3.39     17850.0  United Kingdom

In [11]:
df.shape
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 541909 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    541909 non-null  object 
 1   StockCode    541909 non-null  object 
 2   Description  540455 non-null  object 
 3   Quantity     541909 non-null  int64  
 4   InvoiceDate  541909 non-null  object 
 5   UnitPrice    541909 non-null  float64
 6   CustomerID   406829 non-null  float64
 7   Country      541909 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 33.1+ MB


In [13]:
df[["InvoiceNo", "InvoiceDate", "CustomerID",]].head()


InvoiceNo     InvoiceDate  CustomerID
0    536365  12/1/2010 8:26     17850.0
1    536365  12/1/2010 8:26     17850.0
2    536365  12/1/2010 8:26     17850.0
3    536365  12/1/2010 8:26     17850.0
4    536365  12/1/2010 8:26     17850.0

In [14]:
df.isnull().sum()


InvoiceNo           0
StockCode           0
Description      1454
Quantity            0
InvoiceDate         0
UnitPrice           0
CustomerID     135080
Country             0
dtype: int64

In [17]:
df.duplicated().sum()



np.int64(0)

In [19]:
df = df.drop_duplicates()
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 536641 entries, 0 to 541908
Data columns (total 8 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   InvoiceNo    536641 non-null  object 
 1   StockCode    536641 non-null  object 
 2   Description  535187 non-null  object 
 3   Quantity     536641 non-null  int64  
 4   InvoiceDate  536641 non-null  object 
 5   UnitPrice    536641 non-null  float64
 6   CustomerID   401604 non-null  float64
 7   Country      536641 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 36.8+ MB


In [21]:
df["InvoiceDate"].head()


0   2010-12-01 08:26:00
1   2010-12-01 08:26:00
2   2010-12-01 08:26:00
3   2010-12-01 08:26:00
4   2010-12-01 08:26:00
Name: InvoiceDate, dtype: datetime64[ns]

In [20]:
#converted date type
df["InvoiceDate"] = pd.to_datetime(df["InvoiceDate"], errors="coerce")


In [22]:
#Customer Segmentation
customer_freq = df.groupby("CustomerID")["InvoiceNo"].nunique()


In [26]:
df['Revenue']= df['Quantity']*df['UnitPrice']

In [27]:
customer_spending = df.groupby("CustomerID")["Revenue"].sum()


In [28]:
df["Country"].value_counts()


Country
United Kingdom          490300
Germany                   9480
France                    8541
EIRE                      8184
Spain                     2528
Netherlands               2371
Belgium                   2069
Switzerland               1994
Portugal                  1510
Australia                 1258
Norway                    1086
Italy                      803
Channel Islands            757
Finland                    695
Cyprus                     611
Sweden                     461
Unspecified                442
Austria                    401
Denmark                    389
Japan                      358
Poland                     341
Israel                     294
USA                        291
Hong Kong                  284
Singapore                  229
Iceland                    182
Canada                     151
Greece                     146
Malta                      127
United Arab Emirates        68
European Community          61
RSA                         58
Lebanon                     45
Lithuania                   35
Brazil                      32
Czech Republic              30
Bahrain                     19
Saudi Arabia                10
Name: count, dtype: int64

In [31]:
#ummary of Dataset
df.describe()

Quantity                    InvoiceDate      UnitPrice  \
count  536641.000000                         536641  536641.000000   
mean        9.620029  2011-07-04 08:57:06.087421952       4.632656   
min    -80995.000000            2010-12-01 08:26:00  -11062.060000   
25%         1.000000            2011-03-28 10:52:00       1.250000   
50%         3.000000            2011-07-19 14:04:00       2.080000   
75%        10.000000            2011-10-18 17:05:00       4.130000   
max     80995.000000            2011-12-09 12:50:00   38970.000000   
std       219.130156                            NaN      97.233118   

          CustomerID        Revenue  
count  401604.000000  536641.000000  
mean    15281.160818      18.123861  
min     12346.000000 -168469.600000  
25%     13939.000000       3.750000  
50%     15145.000000       9.870000  
75%     16784.000000      17.400000  
max     18287.000000  168469.600000  
std      1714.006089     380.656263

In [32]:
df.to_csv("cleaned_ecommerce_data.csv", index=False)


Dataset Overview

Contains over 536641 entries

Includes 8 columns

Covers multiple countries

Revenue column created manually

Data Quality Issues

Missing CustomerID values

Duplicate rows present

Some negative Quantity values (returns)


In [33]:
# Remove Critical Missing Values
df = df.dropna(subset=["InvoiceNo", "InvoiceDate", "CustomerID", "Quantity", "UnitPrice"])


In [34]:
#Remove Duplicate Transactions Based on Key Fields
df = df.drop_duplicates(subset=["InvoiceNo", "StockCode", "CustomerID"])


In [36]:
df["Quantity"] = pd.to_numeric(df["Quantity"], errors="coerce")
df["UnitPrice"] = pd.to_numeric(df["UnitPrice"], errors="coerce")
df["CustomerID"] = df["CustomerID"].astype(str)


In [37]:
#Remove Cancelled Orders Invoices starting with "C" indicate cancellations.
df = df[~df["InvoiceNo"].str.startswith("C")]


In [38]:
#Standardize Text Columns
df["Description"] = df["Description"].str.strip().str.lower()
df["Country"] = df["Country"].str.strip().str.title()


In [39]:
#Identify Outliers

Q1 = df["Revenue"].quantile(0.25)
Q3 = df["Revenue"].quantile(0.75)
IQR = Q3 - Q1

lower_bound = Q1 - 1.5 * IQR
upper_bound = Q3 + 1.5 * IQR


In [40]:
#extreame outlier remove
df = df[(df["Revenue"] >= lower_bound) & (df["Revenue"] <= upper_bound)]


In [41]:
df["Year"] = df["InvoiceDate"].dt.year
df["Month"] = df["InvoiceDate"].dt.month
df["Day"] = df["InvoiceDate"].dt.day
df["DayOfWeek"] = df["InvoiceDate"].dt.day_name()


In [43]:
#Create Order-Level Revenue (Optional
order_revenue = df.groupby("InvoiceNo")["Revenue"].sum().reset_index()


In [46]:
df.to_csv("Cleaned_ecommerce_data.csv", index=False)


In [45]:
from google.colab import files
files.download("cleaned_ecommerce_data.csv")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

Final Cleaning Summary

After cleaning, your dataset:

✅ No missing critical values

✅ No duplicate transactions

✅ No cancelled orders

✅ No negative quantity or price

✅ Standardized text

✅ Outliers handled

✅ Revenue column created

✅ Date features extracted